eitprocessing/datahandling/intervaldata.py

```diff
@@ -0,0 +1,185 @@
+    is `False`. If the time range is partially included, the start and end times are trimmed to the start and end time
+    of the selection.
+
+    A potential use case where `partial_inclusion` should be set to `True` is "set_driving_pressure": you might want to
+    keep the driving pressure that was set before the start of the selectioon. A use case where `partial_inclusion`
+    should be set to `False` is "detected_breaths": you might want to ignore partial breaths that started before or
+    ended after the selected period.
+
+    Note that when selecting by time, the end time is included in the selection.
+
+    Args:
+        label: a computer-readable name
+        name: a human-readable name
+        unit: the unit associated with the data
+        category: the category of data
+        time_ranges: a list of time ranges (tuples containing a start time and end time)
+        values: an optional list of values with the same length as time_ranges
+        parameters: parameters used to derive the data
+        derived_from: list of data sets this data was derived from
+        description: extended human readible description of the data
+        partial_inclusion: whether to include a trimmed version of a time range when selecting data
+    """
+
+    label: str
+    name: str
+    unit: str | None
+    category: str
+    time_ranges: list[TimeRange | tuple[float, float]]
+    values: list[Any] | None = None
+    parameters: dict[str, Any] = field(default_factory=dict)
+    derived_from: list[Any] = field(default_factory=list)
+    description: str = ""
+    partial_inclusion: bool = False
+
+    def __post_init__(self) -> None:
+        self.time_ranges = [TimeRange._make(time_range) for time_range in self.time_ranges]
+        self._check_equivalence = ["unit", "category", "parameters", "partial_inclusion"]
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}('{self.label}')"
+
+    def __len__(self) -> int:
+        return len(self.time_ranges)
+
+    def select_by_time(  # noqa: C901
+        self,
+        start_time: float | None = None,
+        end_time: float | None = None,
+        partial_inclusion: bool | None = None,
+        newlabel: str | None = None,
+    ) -> Self:
+        """Return only period data that overlaps (partly) with start and end time.
+
+        Other types of data (e.g. ContinuousData and SparseData) support the start_inclusive and end_inclusive
+        arguments. PeriodData does not. That means that selection by time of PeriodData probably works slightly
+        different than other types of selecting/slicing data.
+        """
+        if partial_inclusion is None:
+            partial_inclusion = self.partial_inclusion
+        newlabel = newlabel or self.label
+
+        if start_time is None:
+            start_time = self.time_ranges[0].start_time
+        if end_time is None:
+            end_time = self.time_ranges[-1].end_time
```

Comment:
I think it would not be beneficial to have the meaning of an argument change with the number of arguments. You can use `data.t[:end_time]` if you want a quick way to only give the end time.